File: 프로그래머스/unrated/155652. 둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    alpha = {}
    answer = ''
    for c in s:
        char = ord(c)
        for i in range(index):
            char += 1
            if (char > ord('z')):
                char = char - ord('z') + ord('a') - 1
            while (chr(char) in skip):
                char += 1
                if (char > ord('z')):
                    char = char - ord('z') + ord('a') - 1
            
        answer += chr(char)
        
    return answer

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.cpp
#include <string>
#include <unordered_map>
#include <iostream>

using namespace std;

string solution(string s, string skip, int index) {
    string answer = "";
    unordered_map<int, bool> skipAlpha;
    
    for (const auto &c : skip) {
        int num = c;
        skipAlpha[c] = true;
    }
    
    for (int i = 0; i < s.size(); i++) {
        int cnt = index;
        int alpha = s[i];
        while (cnt > 0) {
            alpha++;
            if (alpha > 'z') alpha = 'a';
            if (skipAlpha[alpha] == 1) continue;
            cnt--;
        }
        answer += alpha;
    }
    
    return answer;
}
